tests/cpp/test_multi_matmul_scheduler.cpp

```diff
@@ -0,0 +1,620 @@
+      }
+    }
+
+    // Inspect definition. If it's a LoadStoreOp, check that the op type and
+    // cache_op match
+    EXPECT_EQ(
+        tv_new->definition() == nullptr, tv_orig->definition() == nullptr);
+    if (auto* lsop_orig = dynamic_cast<LoadStoreOp*>(tv_orig->definition())) {
+      auto lsop_new = dynamic_cast<LoadStoreOp*>(tv_new->definition());
+      ASSERT_TRUE(lsop_new != nullptr) << suffix;
+      EXPECT_EQ(lsop_new->opType(), lsop_orig->opType());
+      EXPECT_EQ(lsop_new->cacheOp(), lsop_orig->cacheOp());
+    } else if (
+        auto* rop_orig = dynamic_cast<ReductionOp*>(tv_orig->definition())) {
+      auto rop_new = dynamic_cast<ReductionOp*>(tv_new->definition());
+      EXPECT_EQ(
+          rop_new->serialGridReductionRequested(),
+          rop_orig->serialGridReductionRequested());
+    }
+
+    // Print suffix whenever a test fails before recursing
+    ASSERT_FALSE(testing::Test::HasFailure()) << suffix;
+
+    Expr* def_orig = tv_orig->definition();
+    Expr* def_new = tv_new->definition();
+    EXPECT_TRUE(def_orig->sameOp(def_new))
+        << "def_orig\n  " << def_orig->toString()
+        << "is not the same as def_new\n  " << def_new->toString() << suffix;
+    EXPECT_EQ(def_orig->attributes().size(), def_new->attributes().size())
+        << suffix;
+    EXPECT_EQ(def_orig->inputs().size(), def_new->inputs().size()) << suffix;
+    for (size_t i : c10::irange(def_orig->inputs().size())) {
+      if (i >= def_new->inputs().size()) {
+        break;
+      }
+      auto* tv_inp_orig = dynamic_cast<TensorView*>(def_orig->input(i));
+      auto* tv_inp_new = dynamic_cast<TensorView*>(def_new->input(i));
+      EXPECT_EQ(tv_inp_orig != nullptr, tv_inp_new != nullptr) << suffix;
+
+      if (!testing::Test::HasFailure() && tv_inp_orig != nullptr &&
+          !tv_inp_orig->isFusionInput()) {
+        // Skip comparing input TVs since their schedules don't affect the
+        // generated kernel
+        EXPECT_FALSE(tv_inp_new->isFusionInput());
+        compareTVs(tv_inp_orig, tv_inp_new);
+      }
+    }
+
+    // Print suffix whenever a test fails in recursive call
+    ASSERT_FALSE(testing::Test::HasFailure()) << suffix;
+  }
+
+  void compareSchedules() {
+    // clone fusion for scheduling with original matmul scheduler
+    Fusion new_fusion;
+    cloner_ = std::make_unique<IrCloner>(Fusion::copy(fusion, &new_fusion));
+
+    // Schedule fusion with original matmul scheduler
+    scheduleMatmul(fusion, params);
+
+    // Schedule cloned fusion with new scheduler
+    scheduleMultipleMatmuls(&new_fusion, params);
+
+    // find tensors to compare
+    auto getTensorsToCompare = [](Fusion* fusion) {
+      std::vector<TensorView*> tvs;
+
+      for (Val* v : fusion->inputs()) {
+        EXPECT_TRUE(v->isA<TensorView>());
+        // Take the consumer of each input, which is the smem store
+        tvs.push_back(v->uses().at(0)->output(0)->as<TensorView>());
```

Comment:
Currently this just checks the smem store. But `compareTvs` is recursive so we can change this to add the smem loads next, then the MmaOp outputs, and eventually the fusion outputs, in order to check every tensor in the fusion.